src/wrapper.jl

```diff
@@ -465,7 +467,35 @@ function LGBM_BoosterUpdateOneIter(bst::Booster)
     return is_finished[]
 end
 
-# function LGBM_BoosterUpdateOneIterCustom()
+"""
+LGBM_BoosterUpdateOneIterCustom
+Pass grads and 2nd derivatives corresponding to some custom loss function
+grads and 2nd derivatives must be same cardinality as training data
+Also, trying to run this on a booster without data will fail.
+"""
+function LGBM_BoosterUpdateOneIterCustom(bst::Booster, grads::Vector{<:AbstractFloat}, hessian::Vector{<:AbstractFloat})
+
+    if length(bst.datasets) == 0
+        throw(ErrorException("Booster does not have any training data associated"))
+    end
+    numdata = LGBM_DatasetGetNumData(first(bst.datasets))
+
+    if !(numdata == length(grads) == length(hessian))
```

Comment:
Should be equal to `numdata * LGBM_BoosterNumModelPerIteration()`
https://lightgbm.readthedocs.io/en/latest/C-API.html#c.LGBM_BoosterNumModelPerIteration
Please refer to https://github.com/microsoft/LightGBM/pull/4815#issuecomment-974569240.

Reply:
@yaxxie I will wait for this before merging it

Reply:
Yep, I have some modifications to make WRT this.